### PR TITLE
[AACS][Preview] 2024-09-15-preview Azure AI Content Safety

### LINKED ISSUE #30080: [Azure AI Content Safety - Azure AI Content Safety] API Review
New API Review meeting has been requested.

**Service Name**: Azure AI Content Safety - Azure AI Content Safety
**Review Created By**: Bowen Gong
**Review Date**: 08/08/2024 04:00 PM PT
**Release Plan**: [1382](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=ee5b52d2-2553-ef11-a316-6045bd08e35c)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/30073
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=37c151b2-2853-ef11-a317-000d3a3783ac)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by bowgong**
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: https://microsoft-my.sharepoint.com/:w:/p/luhuo/ESNFlU2rVbdEhY8zDt7iB60Bz3wfKM1imSMBcai58fgo6A?e=2jS4oj
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs/pull/28383
* Updated paths: 
  * /text:detectProtectedMaterialForCode - new added in this preview version
  * /imageWithText:analyze - from previous private preview version (reviewed)
  * /text:detectGroundedness - from previous public preview version (reviewed) and add optional properties
  * /text/autoReviewers/* - from previous public preview version (reviewed)
  * /text/customCategory/* - from previous public preview version (reviewed)
  * /image/incidents/* - from previous public preview version (reviewed)
  * /text/incidents/* - from previous public preview version (reviewed)

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

fix https://github.com/Azure/azure-rest-api-specs/issues/30080

### CODE REVIEW

File: specification/cognitiveservices/ContentSafety/examples/2024-09-01/ListTextBlocklistItems.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "operationId": "TextBlocklists_ListTextBlocklistItems",
+  "parameters": {
+    "api-version": "2024-09-01",
+    "blocklistName": "TestBlocklist"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
```
Comment (Sayaka-4987) [inline specification/cognitiveservices/ContentSafety/examples/2024-09-01/ListTextBlocklistItems.json:12]:
```
{
            "blocklistItemId": "9511969e-f1e3-4604-9127-05ee16c509ec",
            "description": "Hate word",
            "text": "hate",
            "isRegex": false
},
{
            "blocklistItemId": "d9b2d63d-a233-4123-847a-7d1b5b3b8a8e",
            "description": "A regular expression that matches harmful words.",
            "text": "b[i1][a@][s\\$]"
            "isRegex": true
}
```

File: specification/cognitiveservices/ContentSafety/examples/2024-09-15-preview/AddOrUpdateBlocklistItems.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "title": "Add or Update BlocklistItems To Text Blocklist",
+  "operationId": "TextBlocklists_AddOrUpdateBlocklistItems",
+  "parameters": {
+    "api-version": "2024-09-15-preview",
+    "blocklistName": "TestBlocklist",
+    "body": {
+      "blocklistItems": [
+        {
```
Comment (Sayaka-4987) [inline specification/cognitiveservices/ContentSafety/examples/2024-09-15-preview/AddOrUpdateBlocklistItems.json:9]: same as 2024-09-01

File: specification/cognitiveservices/ContentSafety/examples/2024-09-15-preview/ListTextBlocklistItems.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "operationId": "TextBlocklists_ListTextBlocklistItems",
+  "parameters": {
+    "api-version": "2024-09-15-preview",
+    "blocklistName": "TestBlocklist"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
```
Comment (Sayaka-4987) [inline specification/cognitiveservices/ContentSafety/examples/2024-09-15-preview/ListTextBlocklistItems.json:12]: same as 2024-09-01

File: specification/cognitiveservices/ContentSafety/models.tsp [hunk 4]
```diff
@@ -193,148 +211,177 @@ model RemoveTextBlocklistItemsOptions {
...
   #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use camelCase"
-  @doc("Deployment model name.")
+  @doc("Deployment name.")
   azureOpenAIDeploymentName: string;
 }
 
-@added(Versions.v2024_02_15_Preview)
-@doc("The request of groundedness detection.")
+@added(ContentSafety.Versions.v2024_09_15_Preview)
+@doc("The request of test groundedness analysis.")
```
Comment (judyzhoubaby) [inline specification/cognitiveservices/ContentSafety/models.tsp:334]: typo: text

Comment (JieZhou000) [inline specification/cognitiveservices/ContentSafety/models.tsp:334]: typo: text

File: specification/cognitiveservices/ContentSafety/models.tsp [hunk 5]
```diff
@@ -193,148 +211,177 @@ model RemoveTextBlocklistItemsOptions {
...
   @doc("""
-    The user's question input in a QnA scenario.
-    This field is optional, but if the task type is set to QnA, it becomes required.
-    """)
+  The user's question input in a QnA scenario.
+  This field is optional, but cannot be omitted when task type is QnA.
+  """)
   qna?: QnAOptions;
 
-  @doc("The specific text that needs to be checked.")
```
Comment (JieZhou000) [inline specification/cognitiveservices/ContentSafety/models.tsp:313]: LLM(Large Language Model)

File: specification/cognitiveservices/ContentSafety/routes.tsp [hunk 6]
```diff
@@ -72,15 +157,48 @@ interface ImageOperations {
...
+@added(ContentSafety.Versions.v2024_09_15_Preview)
+interface ImageWithTextOperations {
+  @summary("Analyze Image with Text")
+  @doc("A synchronous API to analyze image content with embedded or custom text content.")
+  @route("/imageWithText:analyze")
+  @post
+  analyzeImageWithText is Azure.Core.RpcOperation<
+    {
+      @doc("The request body of image-with-text analysis.")
+      @body
```
Comment (johanste) [inline specification/cognitiveservices/ContentSafety/routes.tsp:197]: Why do you wrap this body with `at-body`?

File: specification/cognitiveservices/ContentSafety/models.tsp [hunk 7]
```diff
@@ -265,3 +265,781 @@ model TextProtectedMaterialAnalysisResult {
...
+  @pattern("^(?=[a-zA-Z0-9])[a-zA-Z0-9-]*[a-zA-Z0-9]$")
+  @maxLength(64)
+  operationId?: string;
+}
+
+@added(ContentSafety.Versions.v2024_09_15_Preview)
+@doc("Incident long running operation status result.")
+model IncidentLongRunningOperationStatusResult {
+  @doc("The HTTP status code.")
+  statusCode: 202;
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafety/models.tsp:832]: Please double check.

Comment (bowgong) [reply specification/cognitiveservices/ContentSafety/models.tsp]: I believe this is from some outdated implementation. Just removed it.

File: specification/cognitiveservices/ContentSafety/routes.tsp (hunk 6 above)
Comment (bowgong) [reply specification/cognitiveservices/ContentSafety/routes.tsp:197]: We just follow the previous implementation. Could you please advise what's the best practice here?

**Pull request closed without merging**